build/kubernetes/k8s_setup.sh

```diff
@@ -11,9 +11,12 @@ curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/${KIND
 kind create cluster --image kindest/node:${K8S_VERSION}
 
 # Wait for cluster to be ready
-JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}';
-until kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do
-  sleep 1;
+JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'
+out=''
+until [[ "${out}" =~ 'Ready=True' ]]; do
+  sleep 1
+  out=$(kubectl get nodes -o jsonpath="$JSONPATH")
+  echo "${out}"
```

Comment:
This probably would have made it clear why [this failure occurred](https://github.com/coredns/coredns/pull/7659#issuecomment-3505374573) (_`kubectl` was XML instead of a binary which caused a 10m timeout failure_)